src/framework/core/js/ModelTransformationTransforms.js

```diff
@@ -349,86 +349,188 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
         invertConfiguration: "fluid.transforms.arrayToSetMembership.invert"
     });
 
-
-    fluid.transforms.arrayToSetMembership = function (value, transformSpec, transformer) {
-        var output = {};
-        var options = transformSpec.options;
-
+    /**
+     *
+     * Transform the input array into an object based upon the options provided to the specification. Accepts
+     * an array as its first argument and an optional second argument (an object) which contains the
+     * specifications for the resulting output.
+     *
+     * An error will be thrown if the input to be transformed is not an array.
+     *
+     * Partly invertible (into setMembershipToArray)
+     *
+     * For example, with the following transformSpec:
+     *
+     * {
+     *   "options": {
+     *      "mouse": "hasMouse",
+     *      "keyboard": "hasKeyboard",
+     *      "trackpad": "hasTrackpad",
+     *      "headtracker": "hasHeadtracker"
+     *   },
+     *   "presentValue": "supported",
+     *   "missingValue": "not supported"
+     * }
+     *
+     * For an input array like this:
+     *
+     * ["keyboard", "mouse"]
+     *
+     * The output object will be:
+     *
+     * {
+     *   "hasMouse": "supported",
+     *   "hasKeyboard": "supported",
+     *   "hasTrackpad": "not supported",
+     *   "hasHeadtracker": "not supported"
+     * }
+     *
+     * https://docs.fluidproject.org/infusion/development/ModelTransformationAPI.html#fluidtransformsarraytosetmembership
+     *
+     * @param {Array} value - The Array to be transformed into an object.
+     * @param {Object} [transformSpec] - (optional) The options provided to the transformation rule.
+     * @return {Object} - The transformed object
+     *
+     */
+    fluid.transforms.arrayToSetMembership = function (value, transformSpec) {
+        // Input <value> should be an array
         if (!value || !fluid.isArrayable(value)) {
-            fluid.fail("arrayToSetMembership didn't find array at inputPath nor passed as value.", transformSpec);
+            fluid.fail("arrayToSetMembership didn't find array at inputPath nor passed as value.");
         }
-        if (!options) {
-            fluid.fail("arrayToSetMembership requires an options block set");
-        }
-
-        if (transformSpec.presentValue === undefined) {
-            transformSpec.presentValue = true;
+        var output = {};
+        var presentValue = fluid.get(transformSpec, "presentValue") ? transformSpec.presentValue : true,
+            missingValue = fluid.get(transformSpec, "missingValue") ? transformSpec.missingValue : false,
+            options = fluid.get(transformSpec, "options");
+
+        /**
+         * When <options> are provided in the transformation rule, use them for keys in the output object.
+         * Otherwise, use items from the <value> array as keys.
+         */
+        if (options !== undefined) {
```

Comment:
Feel totally free to ignore this suggestion, but my personal quite mild preference is to always position the positive branch of a condition first - if (option === undefined) { ... even if there is a fairly clear sense, as in this case, that the dominant logical pathway, is the other branch. I wouldn't go so far as the developer who advised this stackexchange questioner - https://softwareengineering.stackexchange.com/questions/350472/developer-insists-if-statements-shouldnt-have-negated-conditions-and-should-al - and create an entire empty branch to satisfy the preference. But if both branches have to be there, I would tend to order them the other way round. If you think the code is clearer as it stands I am perfectly happy to merge it as it is.